rita_common/src/middleware.rs

```diff
@@ -33,55 +65,105 @@ impl<S> Middleware<S> for Headers {
+            }
+
+            if !origin.is_empty() {
+                #[cfg(not(feature = "dash_debug"))]
+                resp.headers_mut().insert(
+                    ACCESS_CONTROL_ALLOW_ORIGIN,
+                    header::HeaderValue::from_str(&format!("http://{}", origin)).unwrap(),
+                );
+                #[cfg(feature = "dash_debug")]
+                resp.headers_mut().insert(
+                    ACCESS_CONTROL_ALLOW_ORIGIN,
+                    header::HeaderValue::from_str("*").unwrap(),
+                );
+            }
             resp.headers_mut().insert(
-                header::HeaderName::try_from("Access-Control-Allow-Origin").unwrap(),
-                header::HeaderValue::from_str("*").unwrap(),
+                ACCESS_CONTROL_ALLOW_HEADERS,
+                header::HeaderValue::from_static("authorization, content-type"),
             );
+
+            Ok(resp)
         }
-        resp.headers_mut().insert(
-            header::HeaderName::try_from("Access-Control-Allow-Headers").unwrap(),
-            header::HeaderValue::from_static("authorization, content-type"),
-        );
-        Ok(Response::Done(resp))
+        .boxed_local()
     }
 }
 
-// for some reason the Headers struct doesn't get this
-#[allow(dead_code)]
-pub struct Auth;
+pub struct AuthMiddlewareFactory;
 
-impl<S> Middleware<S> for Auth {
-    fn start(&self, req: &HttpRequest<S>) -> Result<Started> {
+impl<S, B> Transform<S, ServiceRequest> for AuthMiddlewareFactory
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error> + 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = AuthMiddleware<S>;
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        ok(AuthMiddleware { service })
+    }
+}
+
+pub struct AuthMiddleware<S> {
+    service: S,
+}
+
+impl<S, B> Service<ServiceRequest> for AuthMiddleware<S>
+where
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error> + 'static,
+{
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type Future = LocalBoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    actix_service::forward_ready!(service);
+
+    fn call(&self, req: ServiceRequest) -> Self::Future {
         let password = settings::get_rita_client().network.rita_dashboard_password;
-        let mut config = Config::default();
 
-        // the /exits path is exempted from authenticaiton so that the
-        // checkup.ash cron script can continue to query it without issue
-        if password.is_none() || req.request().path() == "/exits" {
-            return Ok(Started::Done);
-        }
+        let req_path = req.path().to_string();
+
+        let auth = Authorization::<Basic>::parse(&req);
+
+        let fut = self.service.call(req);
+
+        async move {
```

Comment:
please preserve the comment explaining why the exits path is exempt from authentication.

Reply:
fixed